taichi/backends/vulkan/aot_module_builder_impl.cpp

```diff
@@ -9,10 +9,53 @@ namespace taichi {
 namespace lang {
 namespace vulkan {
 
+// Only responsible for reporting device capabilities
+class AotTargetDevice : public Device {
```

Comment:
Very nice design! Maybe put this in a higher level so that GL aot can benefit from this